core/src/test/java/org/apache/calcite/test/SqlFunctionsTest.java

```diff
@@ -1054,6 +1055,54 @@ private void thereAndBack(byte[] bytes) {
       // ok
     }
   }
+  @Test void testPosition() {
+    assertThat(3, is(position("c", "abcdec")));
+    assertThat(3, is(position("c", "abcdec", 2)));
```

Comment:
These tests are all backwards! For example, rather than
```
assertThat(3, is(position("c", "abcdec", 2)));
```
you should write
```
assertThat(position("c", "abcdec", 2), is(3)));
```
Here, `position("c", "abcdec", 2)` is the thing you are testing, and `3` is the expected result. In fact `is(3)` is the assertion about the expected result.

Why does it matter? If the test fails - say the position call returns 57 - then the framework should print 'got 57, expected 3'. Also, the test reads from left-to-right like an English sentence.

Maybe you're familiar with `assertEquals`, and earlier method where you were SUPPOSED to write the arguments backwards. `assertEquals(3, position("c", "abcdec", 2));` would be the correct way to write this test using `assertEquals`.

Reply:
Ah great point! I had thought this felt a little off, thanks!